### PR TITLE
Reminder email when draft enrollment have not changed for 15 days

### PR DESCRIPTION
**Pull request by Isalafont**
- Template email add
- Add an `enrollment_reminder_mailer.rb` class 
- Create a service to `ExtractDraftEnrollmentsToRemind`
- Create a worker to run `ScheduleReminderEmailForDraftEnrollmentsWorker` every day at 08:00 am 
using => [Sidekiq scheduler](https://sidekiq-scheduler.github.io/sidekiq-scheduler/)
- Add Spec
- Modification on Frontend to display message on Enrollment
Capture d’écran 2022-12-09 à 11 59 25

### CODE REVIEW

**Comment by Isalafont** [comment]
[ETQ Demandeur, je reçois un mail de relance après 15j d'inactivité sur un brouillon](https://trello.com/c/bXCWRH4S/929-etq-demandeur-je-re%C3%A7ois-un-mail-de-relance-apr%C3%A8s-15j-dinactivit%C3%A9-sur-un-brouillon)

File: backend/app/workers/reminder_email_finder.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  # include EnrollmentReminderMailer
+
+  attr_reader :enrollment, :user
+
+  def initialize(enrollments)
+    @enrollments = Enrollment.all
+    @user = user
+  end
+
+  def call(enrollments)
```
Comment (skelz0r) [inline backend/app/workers/reminder_email_finder.rb:13]:
```suggestion
  def call
```

File: backend/app/services/extract_draft_enrollments_to_remind.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      .order(:id, "events.created_at")
+  end
+
+  def draft_enrollments_with_last_update_or_create_events
+    enrollments = draft_enrollments
+    events = enrollments.map { |enrollment| enrollment.events.last }.to_a
+    events.reject { |event| event.name == "reminder" }
+  end
+
+  def created_at_updated_at_last_events?
```
Comment (rdubigny) [inline backend/app/services/extract_draft_enrollments_to_remind.rb:28]:
```suggestion
  def last_event_is_create_or_update?
```

File: backend/app/services/extract_draft_enrollments_to_remind.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class ExtractDraftEnrollmentsToRemind
+  attr_reader :enrollment
+
+  def call
+    if created_at_updated_at_last_events?
+      created_at_updated_at_last_events_ids =
+        draft_enrollments_with_last_update_or_create_events.pluck(:enrollment_id).flatten
```
Comment (rdubigny) [inline backend/app/services/extract_draft_enrollments_to_remind.rb:9]: draft_enrollments_with_last_update_or_create_events is called twice, we must improve this.

maybe you can call `draft_enrollments_with_last_update_or_create_events` first then test that it's not an empty array.

File: backend/app/services/extract_draft_enrollments_to_remind.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      Enrollment.find(created_at_updated_at_last_events_ids)
+    end
+  end
+
+  def draft_enrollments
+    Enrollment.where(status: "draft")
+      .where({updated_at: (Time.new(2022, 11, 1).beginning_of_day)..Time.now.end_of_day - 15.days})
```
Comment (rdubigny) [inline backend/app/services/extract_draft_enrollments_to_remind.rb:16]: Can you put `Time.new(2022, 11, 1)` in a constant at the beginning of the file and give it a name ?

ex: REMIND_FROM_DATE

File: backend/app/services/extract_draft_enrollments_to_remind.rb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      .includes(:events)
+      .where.not({events: {name: %w[notify]}})
+      .order(:id, "events.created_at")
+  end
+
+  def draft_enrollments_with_last_update_or_create_events
```
Comment (rdubigny) [inline backend/app/services/extract_draft_enrollments_to_remind.rb:22]: This function returns events not enrollments

```suggestion
  def last_update_or_create_events_for_draft_enrollments
```

**Review by rdubigny — CHANGES_REQUESTED** [review]

File: backend/app/mailers/enrollment_reminder_mailer.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+  def reminder_draft_enrollment_email
+    @target_api_label = data_provider_config["label"]
+    @enrollment = Enrollment.find(params[:enrollment_id])
+    @demandeur_given_name = @enrollment.demandeurs.first.given_name
+    @demadeur_family_name = @enrollment.demandeurs.first.family_name
+    @url = "#{ENV.fetch("FRONT_HOST")}/#{params[:target_api].tr("_", "-")}/#{params[:enrollment_id]}"
+
+    mail(
+      to: enrollment.demandeurs.pluck(:email),
```
Comment (rdubigny) [inline backend/app/mailers/enrollment_reminder_mailer.rb:14]: when I run the job I got an error:

```
2022-12-22T06:34:53.540Z pid=45477 tid=13ft WARN: NameError: undefined local variable or method `enrollment' for #<EnrollmentReminderMailer:0x0000000000ec90>
Did you mean?  @enrollment
2022-12-22T06:34:53.540Z pid=45477 tid=13ft WARN: /opt/apps/datapass-backend/current/app/mailers/enrollment_reminder_mailer.rb:14:in `reminder_draft_enrollment_email'
```

plz write a test that throw this error before fixing it.

**Review by rdubigny — CHANGES_REQUESTED** [review]
This does not seems to works yet.

Also you should never merge master into your feature branches (there was 2 merge like that). Plz use "rebase" instead.

I have rebased this branch (for the second time) so plz do not push force before a "git pull --rebase".

File: backend/app/mailers/enrollment_reminder_mailer.rb (hunk 6 above)
Comment (Isalafont) [reply backend/app/mailers/enrollment_reminder_mailer.rb]: Thank you for fixing the rebase problems. 

I've added some little changes to the test, specially about the  `Timecop.freeze(Time.now.change(year: 2022, month: 12))` to add a fix date.

**Review by rdubigny — APPROVED** [review]

**Pull request merged**